docs/contributing/debugging.md

```diff
@@ -0,0 +1,33 @@
+### Debugging
+
+Debugging java agent can be a challenging task since some instrumentation
+code is directly inlined into target classes and debugger is
+usually not attached early enough to activate breakpoints
+in agent initialization code (`premain`/`agentmain`, `agent-tooling` artifact etc.).
+
+#### Template methods and agent initialization
```

Comment:
I don't think we use the term "template methods" elsewhere. I think "advice methods" may be better.